python/paddle/fluid/contrib/slim/quantization/mkldnn_post_training_strategy.py

```diff
@@ -0,0 +1,104 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import logging
+import six
+import numpy as np
+from .... import core
+from ..core.strategy import Strategy
+
+__all__ = ['MKLDNNPostTrainingQuantStrategy']
+
+logging.basicConfig(format='%(asctime)s-%(levelname)s: %(message)s')
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+
+
+class MKLDNNPostTrainingQuantStrategy(Strategy):
+    """
+    The strategy for Post Training quantization strategy.
+    """
+
+    def __init__(self, int8_model_save_path=None, fp32_model_path=None):
+        """
+        Args:
+            int8_model_save_path(str): The path to save int8 model with fp32 weights.
+                            None means it doesn't save int8 model. defalut: None.
+            fp32_model_path(str): The path to model with fp32 weight. defalut: None.
+        """
+
+        super(MKLDNNPostTrainingQuantStrategy, self).__init__(0, 0)
+        self.int8_model_save_path = int8_model_save_path
+        if fp32_model_path is None:
+            raise Exception("fp32_model_path is None")
+        self.fp32_model_path = fp32_model_path
+
+    def on_compression_begin(self, context):
+        """
+	    Prepare the data and quantify the model
+        """
+
+        super(MKLDNNPostTrainingQuantStrategy,
+              self).on_compression_begin(context)
+        _logger.info('InferQuantStrategy::on_compression_begin')
+
+        #Prepare the Analysis Config
+        infer_config = core.AnalysisConfig("AnalysisConfig")
+        infer_config.switch_ir_optim(True)
+        infer_config.disable_gpu()
+        infer_config.set_model(self.fp32_model_path)
+        infer_config.enable_mkldnn()
+
+        #Prepare the data for calculating the quantization scales
+        warmup_reader = context.eval_reader()
+        if six.PY2:
+            data = warmup_reader.next()
+
+        if six.PY3:
+            data = warmup_reader.__next__()
+
+        num_images = len(data)
+        images = core.PaddleTensor()
+        images.name = "x"
+        images.shape = [num_images, ] + list(data[0][0].shape)
+        images.dtype = core.PaddleDType.FLOAT32
+        image_data = [img.tolist() for (img, _) in data]
+        image_data = np.array(image_data).astype("float32")
+        image_data = image_data.ravel()
+        images.data = core.PaddleBuf(image_data.tolist())
+
+        labels = core.PaddleTensor()
+        labels.name = "y"
+        labels.shape = [num_images, 1]
+        labels.dtype = core.PaddleDType.INT64
+        label_data = [label for (_, label) in data]
+        labels.data = core.PaddleBuf(label_data)
+
+        warmup_data = [images, labels]
+

```

Comment:
Please, make it running on multiple cores by adding
```
infer_config.set_cpu_math_library_num_threads(???)
```
with a proper argument from `FLAGS_OMP_NUM_THREADS` env variable. Otherwise, the test does not utilize multiple cores.